bot/exts/core/leaderboard.py

```diff
@@ -0,0 +1,206 @@
+        )
+
+        self.stop()
+
+    @ui.button(label="Cancel", style=ButtonStyle.grey, row=0)
+    async def cancel(self, _button: ui.Button, interaction: Interaction) -> None:
+        """Logic for if the deployment is not approved."""
+        await interaction.response.send_message(
+            content=":x: Clearing cache aborted!",
+            ephemeral=False,
+        )
+        self.stop()
+
+
+class Leaderboard(commands.Cog):
+    """Cog for getting game leaderboards."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @staticmethod
+    def get_rank(n: int) -> str:
+        """Get the rank in form of emoji (first 3 places) or ordinal number for `n`."""
+        first_3 = {1: ":first_place:", 2: ":second_place:", 3: ":third_place:"}
+        if n in first_3:
+            return first_3[n]
+
+        suffix = ["th", "st", "nd", "rd", "th"][min(n % 10, 4)]
+        if 11 <= (n % 100) <= 13:
+            suffix = "th"
+        return str(n) + suffix
+
+    async def make_leaderboard(
+        self,
+        author: discord.Member,
+        cached_leaderboard: list[RedisCache]
+    ) -> discord.Embed:
+        """Make a discord embed for the current top 10 members in the cached leaderboard."""
+        if len(cached_leaderboard) == 1:
+            game_leaderboard = await cached_leaderboard[0].to_dict()
+            leaderboard = Counter(game_leaderboard)
+        else:
+            leaderboard = Counter()
+            for lb in cached_leaderboard:
+                game_leaderboard = await lb.to_dict()
+                leaderboard += Counter(game_leaderboard)
+
+        top_ten = leaderboard.most_common(10)
+        try:
+            author_rank = list(leaderboard.keys()).index(author.id) + 1
+        except ValueError:
+            author_rank = 0
+
+        lines = []
+        for index, (member_id, score) in enumerate(top_ten, start=1):
+            mention = f"<@{member_id}>"
+            rank = self.ordinal_number(index)
+            buffer = 4 if index >= 4 else len(rank)+1
+            formatted_rank = format(rank, f"\u2800<{buffer}")
+            lines.append(f"{formatted_rank}{score:\u2800<4}— {mention}")
+
+        embed = discord.Embed(
+            title="Top 10",
+            description="\n".join(lines) if lines else "(no entries yet)",
+            colour=Colour.from_rgb(255, 230, 102),
+        )
+        embed.set_footer(
+            text=(
+                f"Page 1/{math.ceil(len(leaderboard) / 10)}  |"
+                f"  Your rank: {author_rank}/{len(leaderboard)}"
+            )
+        )
+        embed.set_thumbnail(url=DUCKY_COINS_THUMBNAIL)
+
+        return embed
+
+    @commands.group(aliases=("lb",), invoke_without_command=True)
+    async def leaderboard(self, ctx: commands.Context, game: ParentCogConvertor = None) -> None:
+        """Get overall leaderboard if not game specified, else leaderboard for that game."""
+        if ctx.invoked_subcommand:
+            return
```

Comment:
This is what `invoke_without_command` accomplishes, no?

Reply:
Probably, not the same but can be used to achieve the same functionality by `if not ctx.invoke_without_command` as `invoke_without_command` gives `True` if there is no subcommand. But `invoked_subcommand` is much simple and explicit IMO.

Reply:
I am talking about the kwarg you pass to `@commands.group()`. This if-statement should be unecessary